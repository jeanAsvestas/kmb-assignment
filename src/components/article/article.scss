.article {
  flex: 0 0 33.333333%;

  .article-image {
    max-height: 200px;
    object-fit: contain;
    margin-right: 10px;
    cursor: pointer;
    width: 100%;
    transition: transform 450ms;
  }
}
