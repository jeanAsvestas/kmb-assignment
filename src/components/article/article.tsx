import './article.scss';
import LocalOfferIcon from '@mui/icons-material/LocalOffer';
import type { ArticleInterface } from '../../model/main.types';

export const Article = (props: { article: ArticleInterface }) => {
  const { article } = props;
  return (
    <div className="article">
      <img
        className="article-image"
        src={
          article.urlToImage ??
          'https://www.shutterstock.com/image-vector/blank-image-placeholder-icon-design-600w-1715514199.jpg'
        }
        alt="poster"
      />
      <h1>{article.publishedAt}</h1>
      <h2>{article.source.name}</h2>
      <p>{article.title}</p>
      <div>
        <LocalOfferIcon />
        <span>${article.author}</span>
      </div>
    </div>
  );
};
