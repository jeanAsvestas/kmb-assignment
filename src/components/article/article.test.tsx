import TestRenderer from 'react-test-renderer';
import { Article } from './article';

it('renders the component as expected', () => {
  const component = TestRenderer.create(<Article article={articles} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

const articles = {
  source: { id: null, name: 'YouTube' },
  author: null,
  title: "Des data centers pour chauffer l'eau des immeubles - Tech & Co",
  description:
    'Le numérique est-il compatible avec la sobriété énergétique ? En région parisienne, l’entreprise Neutral-IT récupère la chaleur dégagée par ses data centers ...',
  url: 'https://www.youtube.com/watch?v=kaocfZmYQ9M',
  urlToImage: 'https://i.ytimg.com/vi/kaocfZmYQ9M/maxresdefault.jpg',
  publishedAt: '2023-01-15T06:01:34Z',
  content: null,
};
