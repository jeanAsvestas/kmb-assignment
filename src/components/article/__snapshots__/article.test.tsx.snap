// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component as expected 1`] = `
<div
  className="article"
>
  <img
    alt="poster"
    className="article-image"
    src="https://i.ytimg.com/vi/kaocfZmYQ9M/maxresdefault.jpg"
  />
  <h1>
    2023-01-15T06:01:34Z
  </h1>
  <h2>
    YouTube
  </h2>
  <p>
    Des data centers pour chauffer l'eau des immeubles - Tech & Co
  </p>
  <div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="LocalOfferIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m21.41 11.58-9-9C12.05 2.22 11.55 2 11 2H4c-1.1 0-2 .9-2 2v7c0 .55.22 1.05.59 1.42l9 9c.36.36.86.58 1.41.58.55 0 1.05-.22 1.41-.59l7-7c.37-.36.59-.86.59-1.41 0-.55-.23-1.06-.59-1.42zM5.5 7C4.67 7 4 6.33 4 5.5S4.67 4 5.5 4 7 4.67 7 5.5 6.33 7 5.5 7z"
      />
    </svg>
    <span>
      $
    </span>
  </div>
</div>
`;
